### ISSUE TITLE
undefined method `logger' for Digidoc::Client:Class

### ISSUE THREAD

**Issue by whysthatso** [tarmotalu/digidoc_client#3]
i get this error when using the gem in my rails app

when i comment out logger: Client.logger in client.rb, line 325, all goes well.

any ideas?

**Comment by tarmotalu** [comment]
It uses Savon logger, try add initializer like this:

```ruby
Savon.configure do |savon|
  savon.logger = Logger.new('log/webservices.log')
  savon.raise_errors = false
end
```

**Comment by tarmotalu** [comment]
I'm sorry, I copied configuration lines for the old version.

You should create initializer for Digidoc client like this:
```ruby
Digidoc::Client.logger = Logger.new('log/webservices.log')
```

**Comment by whysthatso** [comment]
still not sure if i did it right.

right now whenever i need a Digidoc::Client object, i create it like so:

```
Digidoc::Client.new.logger = Logger.new('log/mobile_id_service.log')
```

rather than your proposed 

```
Digidoc::Client.logger = Logger.new('log/webservices.log')
```

which gave me a noMethodError

Does that make sense? And does it make sense that i initialize the logger whenever i create a new object, or should i "configure" that class somewhere? (i.e. config/initializer/)

**Comment by tarmotalu** [comment]
Hi,

I pushed new changes and relased new version where this issue has been fixed.

https://github.com/tarmotalu/digidoc_client/blob/master/lib/digidoc/client.rb#L338

Best,
Tarmo

On 14.10.2015, at 21:46, Andreas Wagner notifications@github.com wrote:

still not sure if i did it right.

right now whenever i need a Digidoc::Client object, i create it like so:

Digidoc::Client.new.logger = Logger.new('log/mobile_id_service.log')
rather than your proposed

Digidoc::Client.logger = Logger.new('log/webservices.log')
which gave me a noMethodError

Does that make sense? And does it make sense that i initialize the logger whenever i create a new object, or should i "configure" that class somewhere? (i.e. config/initializer/)

**Comment by tarmotalu** [comment]
And while creating a client one way is to use this approach:

...
  def digidoc_client
    @digidoc_client ||= begin
      client = Digidoc::Client.new(“endpoint_url")
      client.logger = Logger.new('log/digidoc_service.log')
      client
    end
  end
...

Tarmo

On 14.10.2015, at 21:55, Tarmo Talu tarmo.talu@wisesoft.ee wrote:

Hi,

I pushed new changes and relased new version where this issue has been fixed.

https://github.com/tarmotalu/digidoc_client/blob/master/lib/digidoc/client.rb#L338 https://github.com/tarmotalu/digidoc_client/blob/master/lib/digidoc/client.rb#L338

Best,
Tarmo

On 14.10.2015, at 21:46, Andreas Wagner <notifications@github.com <mailto:notifications@github.com>> wrote:

still not sure if i did it right.

right now whenever i need a Digidoc::Client object, i create it like so:

Digidoc::Client.new.logger = Logger.new('log/mobile_id_service.log')
rather than your proposed

Digidoc::Client.logger = Logger.new('log/webservices.log')
which gave me a noMethodError

Does that make sense? And does it make sense that i initialize the logger whenever i create a new object, or should i "configure" that class somewhere? (i.e. config/initializer/)
